libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -1985,6 +1984,25 @@ bool WaveTrack::CloseLock() noexcept
    return true;
 }
 
+ClipConstHolders WaveTrack::GetClipInterfaces() const
+{
+   assert(IsLeader());
+   const auto pOwner = GetOwner();
+   assert(pOwner);
+   ClipConstHolders wideClips;
+   wideClips.reserve(mClips.size());
+   for (auto clipIndex = 0u; clipIndex < mClips.size(); ++clipIndex)
+   {
+      const auto leftClip = mClips[clipIndex];
+      const auto rightClip =
+         NChannels() == 2u && pOwner ?
+            (*++pOwner->Find<const WaveTrack>(this))->mClips[clipIndex] :
+            nullptr;
```

Comment:
You handle un-owned tracks anyway, so weaken the `@pre`.

Now remember too, that clips are not necessarily stored, sorted by start time.  But are you assuming that where you use the results of this function?

Reply:
Ok for the `@pre`.
The stretching decorator sorts the clips this method returns, so we're good.

Reply:
Then perhaps the lack of promise of sortedness should be mentioned in a header comment.